### PR TITLE
Add support for monitoring with Prometheus

### PR DESCRIPTION
**Pull request by ijdickinson**
This PR addresses epimorphics/hmlr-linked-data#57 by adding support for Prometheus monitoring on the `/metrics` endpoint.

Internally, we set up `ActiveSupport::Notification` events to record events of interest, then listen to those events to add data to the Prometheus metrics. In addition, the standard Rack middleare provided by the Ruby Prometheus client adds a number of basic HTTP metrics automatically.

- Add Prometheus dependencies
- Inject Rack middleware for Prometheus
- Instrument internal errors
- Add the core machinery for Prometheus monitoring
- Add Prometheus setup for local testing
- Update the README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,17 +45,75 @@ To bypass the need for running locally AWS you can pass a global variable to the
...
+Using this approach, and assuming you install your local copy of
+Prometheus into `~/apps`, a starting command line would be something like:
+
+```sh
+~/apps/prometheus/prometheus-2.32.1.linux-amd64/prometheus \
+  --config.file=test/prometheus/dev.yml \
+  --storage.tsdb.path=./tmp/metrics2
+```
+
+Something roughly equivalent should be possible on Windows and Mac as well.
```
Comment (andrew-pickin-epi) [inline README.md:119]: I'm not sure this is necessary to star up a prometheus server in order to test the metrics I've certainly never done it for images I've produced, I simple curl has been sufficient. But now it's here then great.

**Review by andrew-pickin-epi — APPROVED** [review]

**Review by bogdanadrianmarc — APPROVED** [review]
Lgtm! Is this a big enough change to require a version bump and a changelog entry?

**Review by joescottdave — APPROVED** [review]
Lgtm

**Comment by ijdickinson** [comment]
@bogdanadrianmarc Good point. I've updated the changelog and version

**Pull request merged**